Signal/test/util/ByteParserTest.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import XCTest
+@testable import Signal
+
+class ByteParserTest: XCTestCase {
+
+    override func setUp() {
+        super.setUp()
+//         self.imageCache = ImageCache()
+//        imageCache.setImage(firstVariation, forKey:cacheKey1, diameter:100)
+//        imageCache.setImage(secondVariation, forKey:cacheKey1, diameter:200)
+//        imageCache.setImage(otherImage, forKey:cacheKey2, diameter:100)
+    }
+
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+
+    func testGetShort_Empty() {
+        let parser = ByteParser(data: Data(), littleEndian: true)
+        XCTAssertNotNil(parser)
+        XCTAssertFalse(parser.hasError)
+
+        XCTAssertEqual(0, parser.nextShort())
+        XCTAssertTrue(parser.hasError)
+    }
+
+    func testGetShort() {
+        let data = Data(bytes: [0x01, 0x00, 0x00, 0x01, 0x01, 0x01 ])
+        let parser = ByteParser(data: data, littleEndian: true)
```

Comment:
a `littleEndian: false` test might be nice.

Reply:
Done.